marconi/src/Marconi/CLI.hs

```diff
@@ -69,3 +77,72 @@ parseCardanoAddresses =  nub
     . words
     where
         deserializeToCardano = C.deserialiseFromBech32 (C.proxyToAsType Proxy)
+
+
+-- | This executable is meant to exercise a set of indexers (for now datumhash -> datum)
+--     against the mainnet (meant to be used for testing).
+--
+--     In case you want to access the results of the datumhash indexer you need to query
+--     the resulting database:
+--     $ sqlite3 datums.sqlite
+--     > select slotNo, datumHash, datum from kv_datumhsh_datum where slotNo = 39920450;
+--     39920450|679a55b523ff8d61942b2583b76e5d49498468164802ef1ebe513c685d6fb5c2|X(002f9787436835852ea78d3c45fc3d436b324184
+
+data Options = Options
+  { optionsSocketPath      :: String,
+    optionsNetworkId       :: NetworkId,
+    optionsChainPoint      :: ChainPoint,
+    optionsDbPath          :: FilePath,    -- ^ SQLite database directory path
+    optionsDisableUtxo     :: Bool,
+    optionsDisableDatum    :: Bool,
+    optionsDisableScript   :: Bool,
+    optionsTargetAddresses :: Maybe TargetAddresses
+  }
+  deriving (Show)
+parseOptions :: IO Options
+parseOptions = Opt.execParser $ Opt.info (optionsParser Opt.<**> Opt.helper) mempty
+
+optionsParser :: Opt.Parser Options
+optionsParser =
+  Options
+    <$> Opt.strOption (Opt.long "socket-path" <> Opt.help "Path to node socket.")
+    <*> pNetworkId
+    <*> chainPointParser
+    <*> Opt.strOption (Opt.long "database-directory-path"
```

Comment:
Can we have a short version for this, please :)

Reply:
Here is what they look like now:
`Usage: marconi (-s|--socket-path ARG) (--mainnet | --testnet-magic NATURAL)
               [(-n|--slot-no SLOT-NO) (-b|--block-hash BLOCK-HASH)]
               (-d|--database-directory-path ARG) [--disable-utxo]
               [--disable-datum] [--disable-script-tx]
               [(-a|--addresses-to-index ARG)]

Available options:
  -s,--socket-path ARG     Path to node socket.
  --mainnet                Use the mainnet magic id.
  --testnet-magic NATURAL  Specify a testnet magic id.
  -d,--database-directory-path ARG
                           Dirctory Path for SQLite database.
  --disable-utxo           disable utxo indexers.
  --disable-datum          disable datum indexers.
  --disable-script-tx      disable script-tx indexers.
  -a,--addresses-to-index ARG
                           Becch32 Shelley addresses to index. i.e
                           "--address-to-index address-1 --address-to-index
                           address-2 ..."
  -h,--help                Show this help text`